lib/ddtrace/context.rb

```diff
@@ -161,6 +182,50 @@ def attach_sampling_priority
       )
     end
 
+    # Return the start time of the root span, or nil if there are no spans or this is undefined.
+    def start_time
```

Comment:
can we make these methods "private" or anyway use a convention to avoid people using them? since it's internal machinery, I don't want that developers use an API that could be removed later. If we use the `__start_time`, let's also add in the comment that this is an internal API.